### PR TITLE
[FLINK-29278] BINARY type is not supported in table store

### PR DESCRIPTION
**Pull request by SteNicholas**
`BINARY` type is not supported in table store for `OrcFileFormat`.

**The brief change log**
- Adds the `BINARY` type support for `OrcFileFormat`.

### CODE REVIEW

File: flink-table-store-format/src/main/java/org/apache/flink/table/store/format/orc/OrcFileFormat.java [hunk 1]
```diff
@@ -108,4 +113,71 @@ private static Properties getOrcProperties(ReadableConfig options) {
...
+                }
+            case BOOLEAN:
+                return TypeDescription.createBoolean();
+            case BINARY:
+                return TypeDescription.createBinary();
+            case VARBINARY:
+                if (type.equals(DataTypes.BYTES().getLogicalType())) {
+                    return TypeDescription.createBinary();
+                } else {
+                    throw new UnsupportedOperationException(
```
Comment (JingsongLi) [inline flink-table-store-format/src/main/java/org/apache/flink/table/store/format/orc/OrcFileFormat.java:138]: We can support this too

File: flink-table-store-common/src/test/java/org/apache/flink/table/store/format/FileStatsExtractorTestBase.java [hunk 2]
```diff
@@ -128,6 +129,9 @@ private Object createField(LogicalType type) {
                         randomString(random.nextInt(varCharType.getLength()) + 1));
             case BOOLEAN:
                 return random.nextBoolean();
+            case BINARY:
```
Comment (JingsongLi) [inline flink-table-store-common/src/test/java/org/apache/flink/table/store/format/FileStatsExtractorTestBase.java:132]: I think we should add a case in `FileStoreTableTestBase` to test read and write.

Comment (SteNicholas) [reply flink-table-store-common/src/test/java/org/apache/flink/table/store/format/FileStatsExtractorTestBase.java:133]: @JingsongLi, the `testReadFilter` in `FileStoreTableTestBase` could cover the test case to test read and write. PTAL.

Comment (JingsongLi) [reply flink-table-store-common/src/test/java/org/apache/flink/table/store/format/FileStatsExtractorTestBase.java:133]: We can add test for new types.

File: flink-table-store-format/src/main/java/org/apache/flink/table/store/format/orc/OrcFileFormat.java [hunk 3]
```diff
@@ -92,14 +98,17 @@ public BulkFormat<RowData, FileSourceSplit> createReaderFactory(
...
+                Projection.of(projection).toTopLevelIndexes(),
+                orcPredicates);
     }
 
     @Override
     public BulkWriter.Factory<RowData> createWriterFactory(RowType type) {
         LogicalType[] orcTypes = type.getChildren().toArray(new LogicalType[0]);
 
-        TypeDescription typeDescription = OrcSplitReaderUtil.logicalTypeToOrcType(type);
+        TypeDescription typeDescription = logicalTypeToOrcType(type);
```
Comment (JingsongLi) [inline flink-table-store-format/src/main/java/org/apache/flink/table/store/format/orc/OrcFileFormat.java:111]: can we use `OrcSplitReaderUtil.logicalTypeToOrcType(convertLogicalType(type))`?

File: flink-table-store-format/src/main/java/org/apache/flink/table/store/format/orc/OrcFileFormat.java [hunk 4]
```diff
@@ -115,4 +124,96 @@ private static Properties getOrcProperties(ReadableConfig options) {
         properties.forEach((k, v) -> orcProperties.put(IDENTIFIER + "." + k, v));
         return orcProperties;
     }
+
+    private static LogicalType convertLogicalType(LogicalType type) {
```
Comment (JingsongLi) [inline flink-table-store-format/src/main/java/org/apache/flink/table/store/format/orc/OrcFileFormat.java:128]: can we change a method name? or add some comments to explain `BINARY` and `VARBINARY`

**Review by JingsongLi — APPROVED** [review]
Looks good to me!
Thanks for the update!

**Pull request merged**